Components/Search.js

```diff
@@ -0,0 +1,97 @@
+import React, {useState,useEffect} from "react";  
+import { SafeAreaView ,StyleSheet, View,Text,TextInput} from 'react-native';
+import { FlatList } from "react-native-gesture-handler";
+import { ListItem, Avatar } from 'react-native-elements'
+
+import list from "./Movielist";
+
+
+const SearchScreen=() => {
+
+  const [filterdData,setfilterdData]= useState([]);
+  const [masterData,setmasterData]= useState([]);
+  const [search,setsearch]= useState('')
+
+  useEffect(() =>{
+   fetchPosts();
+    return()=>{
+
+    }
+  },[])
+
+  const fetchPosts =() =>{
+    setfilterdData(list);
+    setmasterData(list);
+  }
+
+  const searchFilter =(text) =>{
+    if(text)
+    {
+        const newData = masterData.filter((item)=>{
+        const itemData= item.title ? item.title.toUpperCase() : ''.toUpperCase();
+        const textdata= text.toUpperCase();
+        return itemData.indexOf(textdata) > -1;
+      });
+      setfilterdData(newData);
+      setsearch(text);
+    }else{
+      setfilterdData(masterData);
+      setsearch(text);
+    }
+  }
+
+  const ItemView = ({item}) =>{
+    return(
+     
+     <ListItem>
+     <Avatar source={{uri: item.avatar_url}} />
+       <ListItem.Content>
+       <ListItem.Title>{item.title}</ListItem.Title>
+       <ListItem.Subtitle>{item.year}</ListItem.Subtitle>
+     </ListItem.Content>
+     </ListItem>
```

Comment:
list item is duplicated in both search and list page. This duplication is a bad idea considering they both are using the same view

Reply:
I will merge both of them through one function